bin.src/plot_astrometric_residuals.py

```diff
@@ -0,0 +1,281 @@
+    nx = 20
+    ny = 40
+    uMean = np.zeros((nx-1, ny-1))
+    vMean = np.zeros((nx-1, ny-1))
+    xMean = np.zeros((nx-1, ny-1))
+    yMean = np.zeros((nx-1, ny-1))
+    ww = np.linspace(xlim[0], xlim[1], nx)
+    hh = np.linspace(ylim[0], ylim[1], ny)
+    for i, (w0, w1) in enumerate(zip(ww[:-1], ww[1:])):
+        inx = (xx >= w0) & (xx <= w1)
+        for j, (h0, h1) in enumerate(zip(hh[:-1], hh[1:])):
+            iny = (yy >= h0) & (yy <= h1)
+            inside = inx & iny
+            xMean[i, j] = (w0 + w1)/2
+            yMean[i, j] = (h0 + h1)/2
+            uMean[i, j] = np.mean(uu[inside])
+            vMean[i, j] = np.mean(vv[inside])
+
+    return xMean, yMean, uMean, vMean
+
+
+def plot_quiver(xx, yy, uu, vv, ccd, label, focal):
+    """Make a quiver plot of the astrometry error vectors."""
+    cycle = plt.rcParams['axes.prop_cycle'].by_key()['color']
+
+    color = cycle[0] if 'jointcal' in label else cycle[1]
+    scale = 0.25 if 'mean' in label else 1
+    width = 7.0
+    if focal:
+        width = .1
+        scale *= 70
+
+    fig = plt.figure(figsize=(6, 10))
+    ax = fig.add_subplot(111)
+
+    # units of the vectors are not, in fact "x", but it's simpler to treat them that way.
+    # uu, vv should be in arcsec, xx,yy in pixels or mm (if focal)
+    Q = ax.quiver(xx, yy, uu, vv, units='x', pivot='tail', scale=scale, width=width,
+                  headwidth=4, clip_on=False, color=color)
+    length = 5/scale if 'mean' in label else 100
+    key = ax.quiverkey(Q, 0.85, 0.90, length, '%s mas'%(length), angle=45,
+                       coordinates='figure', labelpos='W', fontproperties={'size': 24})
+
+    ax.set_xticklabels([])
+    ax.set_yticklabels([])
+
+    plt.title('{}'.format(ccd))
+    filename = "plots/quiver-%s-%s.png"%(ccd, label)
+    plt.savefig(filename, bbox_extra_artists=(key,))
+    plt.close(fig)
+
+
+def main():
+    parser = lsst.pipe.base.InputOnlyArgumentParser("plot_astrometric_residuals",
+                                                    description=__doc__)
+    parser.add_argument("--jointcal", action='store_true',
+                        help="Use the jointcal_wcs dataset to update each ccd's coordinates.")
+    parser.add_argument("--focalplane", action='store_true',
+                        help="Make plots and pickle files in focal plane coordinates.")
+    # We need to specify "jointcal_wcs" here so that the butler understands what a `tract` is.
+    parser.add_id_argument(name="--id", datasetType="jointcal_wcs",
+                           help="data IDs, e.g. --id visit=12345 ccd=1^2 tract=1234")
+
+    args = parser.parse_args(config=lsst.pex.config.Config())
+    name = 'jointcal' if args.jointcal else 'single'
+    butler = args.butler
+
+    if len(args.id.refList) == 0:
+        raise RuntimeError("No data to process! Check your id list and/or input data path.")
+    fluxField, newSchema, mapper, multiMatch = prep_matching(butler, args.id.refList[0])
```

Comment:
Separate this function here where the stuff above is `parse`
and the lines below are `run`.